.github/workflows/welcome_message.yml

```
@@ -0,0 +1,18 @@
+name: 'Welcome New Contributors'
+
+on:
+  issues:
+    types: [opened]
+  pull_request_target:
+    types: [opened]
+
```

It's a best practice to restrict the permissions granted to the `GITHUB_TOKEN` to just the minimum set that are relevant to the workflow/job:

```suggestion

permissions:
  issues: write
  pull-requests: write

```